File: 5 - Form Template Method/JS/CapitalStrategyAdvisedLine.js
import CapitalStrategy from "./CapitalStrategy";

export default class CapitalStrategyAdvisedLine extends CapitalStrategy{
    capital(loan) {
        return loan.getCommitment() * loan.getUnusedPercentage() * this.duration(loan) * this.riskFactor(loan)
    }
}

File: 4 - Replace Conditional Logic with Strategy/C#/MPG.ReplaceConditionalLogicWithStrategy/MPG.ReplaceConditionalLogicWithStrategy.After/CapitalStrategyAdvisedLine.cs
﻿namespace MPG.ReplaceConditionalLogicWithStrategy.After
{
    public class CapitalStrategyAdvisedLine : CapitalStrategy
    {
        public override double Capital(Loan loan)
        {
            return loan.GetCommitment() * loan.GetUnusedPercentage() * loan.Duration() * RiskFactor(loan);
        }
    }
}